### PR TITLE
Reset EntityTreeSendThread known state if client disconnects from Entity Server

### PR DESCRIPTION
**Pull request by birarda**
- clear known state in EntityTreeSendThread if querying client believes it is starting a new connection

### CODE REVIEW

**Comment by birarda** [comment]
This will need to wait until #11811 is in since it includes changes from that.

File: libraries/octree/src/OctreeQuery.h [hunk 1]
```diff
@@ -27,11 +27,11 @@ class OctreeQuery : public NodeData {
     Q_OBJECT
 
 public:
-    OctreeQuery();
+    OctreeQuery(bool randomizeConnectionID = false);
     virtual ~OctreeQuery() {}
 
     int getBroadcastData(unsigned char* destinationBuffer);
-    int parseData(ReceivedMessage& message) override;
+    virtual int parseData(ReceivedMessage& message) override;
```
Comment (huffman) [inline libraries/octree/src/OctreeQuery.h]: unintentionally added virtual?

File: libraries/octree/src/OctreeQuery.h [hunk 2]
```diff
@@ -68,6 +68,14 @@ class OctreeQuery : public NodeData {
     bool getUsesFrustum() { return _usesFrustum; }
     void setUsesFrustum(bool usesFrustum) { _usesFrustum = usesFrustum; }
 
+    void setConnectionID(uint16_t connectionID) { _connectionID = connectionID; }
```
Comment (huffman) [inline libraries/octree/src/OctreeQuery.h]: Needed?

Comment (birarda) [reply libraries/octree/src/OctreeQuery.h]: Nope, was doing the randomization in Application at first and then moved it - I can remove this.

File: libraries/octree/src/OctreeQuery.h (hunk 1 above)
Comment (birarda) [reply libraries/octree/src/OctreeQuery.h]: Yup, was a holdover from the previous way I had the boolean for having received the first query

**Comment by birarda** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4996/HighFidelity-Beta-PR11824-6d5b528cfbd0ab10216d9a19b22e05c6c4ea873e.dmg

**Comment by birarda** [comment]
#### Test Plan

You'll need [clumsy](https://jagt.github.io/clumsy/) to test this.

1. Using this PR, connect to "sport". Confirm that everything loads as you expect (especially the floor).
2. Then, using clumsy, filter for `udp and (udp.SrcPort == 55555)`. Turn on dropping for 100% of inbound packets.
3. Wait for Interface to show a -1 in your stats for Entity Server Ping.
4. Once you see -1, turn off clumsy.
5. Confirm that you see the ping go back to a positive value, and that you see the entities from the domain start to load when you reconnect.
6. Repeat steps 1-4 with master or stable on a domain that it is protocol compatible. You should see that entities do not come back as they did in sport after you turn off clumsy (allowing you to reconnect to the entity server).

**Comment by birarda** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/5000/HighFidelity-Beta-PR11824-6d5b528cfbd0ab10216d9a19b22e05c6c4ea873e.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/5000/HighFidelity-Beta-PR11824-6d5b528cfbd0ab10216d9a19b22e05c6c4ea873e.dmg

**Comment by ZappoMan** [comment]
I'm all for this change. Stephen and I reviewed the design, and I think it's a great fix to this problem. But I don't understand why this is a release blocker? Do we have regular consistent user repros of this problem happen?

**Comment by divyaelango88** [comment]
@ZappoMan yes this has a consistentent 100% repro and is a regression, as you can see the details in this fogbugz ticket - https://highfidelity.fogbugz.com/f/cases/8832/Assets-do-not-reload-after-network-disconnect. It parent issue https://highfidelity.fogbugz.com/f/cases/8903/A-few-users-are-reporting-the-floor-not-loading has been reported by greeters a lot last week as well.

**Pull request closed without merging**